### PR TITLE
Video derivatives

### PR DESCRIPTION
**Pull request by mkorcy**
@tonyc @jcoyne This is the rest of the video derivatives code, I see in travis you guys have 1 failing test, this branch fails that same test.  I'm guessing you may want these tests excluded on travis because they require ffmpeg?

### CODE REVIEW

**Comment by tonyc** [comment]
@mkorcy - I don't think we should skip any tests if at all possible. Are the ENOENT failures in Travis here due to ffmpeg missing? The failures aren't exactly clear.

I'm also a bit dicey seeing filesystem operations in the specs - I'd like to see something like FakeFS (https://github.com/defunkt/fakefs) implemented for these specs.

File: spec/job/create_derivatives_spec.rb [hunk 1]
```diff
@@ -13,7 +13,62 @@
...
+    end
+
+    before(:each) do
+      subject.datastreams["Archival.video"].dsLocation = "http://bucket01.lib.tufts.edu/data01/tufts/central/dca/MISS/archival_video/sample.mp4"
+      subject.datastreams["Archival.video"].mimeType = "video/mp4"
+      subject.save
+    end
+
+
+      it 'raises an error if it the archival video doesn' 't exist' do
```
Comment (tonyc) [inline spec/job/create_derivatives_spec.rb]: Fix the single quoting here, and indentation

File: spec/job/create_derivatives_spec.rb [hunk 2]
```diff
@@ -13,7 +13,62 @@
...
+      webm_path = LocalPathService.new(subject, 'Access.webm').local_path
+      FileUtils.mkdir_p(File.dirname(webm_path))  # in case the derivatives folder doesn't already exist
+      FileUtils.chmod(0444, File.dirname(webm_path))
+
+      expect{job.perform}.to raise_error(Errno::EACCES)
+
+      FileUtils.chmod(0755, File.dirname(webm_path))
+    end
+
+    it 'creates derivatives' do
```
Comment (tonyc) [inline spec/job/create_derivatives_spec.rb]: Consider adding FakeFS around this spec - I'm a bit nervous when a spec starts removing entire directories to clean-up. See https://github.com/defunkt/fakefs#dont-fake-the-fs-immediately

**Comment by mkorcy** [comment]
@tonyc I'll rework using FakeFS, if I don't get to it today, I'm going to be out of the office for 2 weeks so its not that I'm ignoring the request to change this.

**Comment by tonyc** [comment]
Ok, no worries - if you don't get to it by today, I'll merge it in and make the changes myself. Shouldn't be too terribly bad :)

**Comment by tonyc** [comment]
@mkorcy  - it turns out it was not a filesystem issue, but the specs were trying to access bucket01.lib.tufts.edu, which I believe is accessible via VPN only.

I've merged the PR in, but marked the specs as "skip" for the time being until you return.

A couple thoughts:

* These specs actually should live in tufts_models, because it's testing specific behavior of TuftsVideo's ```create_derivatives``` methods.
* The specs should not rely on a VPN connection. There might be a better way to attach a sample file, but I'm not entirely clear how to do it at this moment.

**Comment by tonyc** [comment]
I've since re-enabled the specs, and found that an archival.mp4 was missing, which I've added. I've also updated the README to include a homebrew command for installing ffmpeg with the needed libraries included.

I've also shored up the ENOACCES/ENOENT errors in tufts_models with the paths that were attempted to be accessed.

**Pull request merged**